app/controllers/avo/application_controller.rb

```diff
@@ -257,7 +258,13 @@ def render_unauthorized(exception)
     end
 
     def set_authorization
-      @authorization = Services::AuthorizationService.new _current_user
+      # We need to set @resource_name for the #resource method to work properly
+      @resource_name = resource_name
```

Comment:
this exact line is done on `before_action :set_resource_name`

Reply:
... but it doesn't work :( When I tried debugging it at this step, `@resource_name` would always be nil, despite the `set_resource_name` being called before `set_authorization`. Hence the duplication

Reply:
![image](https://user-images.githubusercontent.com/69730720/188807660-04406a9f-c89b-494c-b39f-3ae4bb48a1c5.png)

Reply:
how? :O

Reply:
yes, i'll investigate that later, for now can we use the `set_resource_name` instead of `@resource_name = resource_name` ? It's easier to maintain this way